apps/testingaccessibility/src/lib/prisma-api.ts

```diff
@@ -15,6 +14,54 @@ export function getSdk(
   {ctx = defaultContext, spanContext}: SDKOptions = {ctx: defaultContext},
 ) {
   return {
+    async toggleLessonProgressForUser({
+      userId,
+      lessonSlug,
+    }: {
+      userId: string
+      lessonSlug: string
+    }) {
+      const span = startSpan('toggleLessonProgressForUser', spanContext)
+      let lessonProgress = await ctx.prisma.lessonProgress.findFirst({
+        where: {
+          userId,
+          lessonSlug,
+        },
+      })
+
+      const now = new Date()
+
+      if (lessonProgress) {
+        if (lessonProgress.completedAt) {
+          lessonProgress = await ctx.prisma.lessonProgress.update({
+            where: {id: lessonProgress.id},
+            data: {
+              completedAt: null,
+              updatedAt: now,
+            },
+          })
+        } else {
+          lessonProgress = await ctx.prisma.lessonProgress.update({
+            where: {id: lessonProgress.id},
+            data: {
+              completedAt: now,
+              updatedAt: now,
+            },
+          })
+        }
+      } else {
+        lessonProgress = await ctx.prisma.lessonProgress.create({
+          data: {
+            userId,
+            lessonSlug,
+            completedAt: now,
+            updatedAt: now,
+          },
+        })
+      }
+      span.finish()
+      return lessonProgress
```

Comment:
this is just the single lesson progress not the entire user progress so this probably changes the logic